### PR TITLE
#2677 docu, how to publish prometheus metric endpoint on all interfaces

### PR DESCRIPTION
**Pull request by mmethner**
Hi @abuehrle, Hi @bboreham  

I added a short explanation on publish metrics on all interfaces throughout the cluster.

Ticket is https://github.com/weaveworks/weave/issues/2677

It's the first time I submit a PR to an mostly unknown Repo. Do I need to add more information or something?

### CODE REVIEW

File: site/metrics.md [hunk 1]
```diff
@@ -20,6 +20,15 @@ exposed:
 * `weave_dns_entries` - Number of DNS entries.
 * `weave_flows` - Number of FastDP flows.
 
+#### Publish Router Metrics Endpoint
+
+By default weave only listens on it's local interface to serve metrics.
```
Comment (bboreham) [inline site/metrics.md]: This is true when started from the `weave` script, but not true when started from Kubernetes via `weave-kube`.

Suggest "By default, when started via `weave launch`"

File: site/metrics.md [hunk 2]
```diff
@@ -20,6 +20,15 @@ exposed:
 * `weave_dns_entries` - Number of DNS entries.
 * `weave_flows` - Number of FastDP flows.
 
+#### Publish Router Metrics Endpoint
+
+By default weave only listens on it's local interface to serve metrics. 
+To publish your metrics throughout your cluster, e.g. if
+your prometheus server is installed on a different host machine,
+you need to set `WEAVE_STATUS_ADDR` to listen on all interfaces:
```
Comment (bboreham) [inline site/metrics.md]: If a user has multiple interfaces they may not not want to listen on interfaces exposed to the public Internet.  I don't think we want to flatly state they should set it to listen on all interfaces.

File: site/metrics.md (hunk 1 above)
Comment (bboreham) [inline site/metrics.md]: it's -> its

File: site/metrics.md [hunk 3]
```diff
@@ -20,6 +20,15 @@ exposed:
...
+`WEAVE_STATUS_ADDR=0.0.0.0:6782`
```
Comment (bboreham) [inline site/metrics.md]: Also they may want to change the port number, if 6782 clashes with another port in use on their machine.

**Comment by mmethner** [comment]
Thanks, that you took the time to help me to be more secure and detailed on this part of the documentation. I updated the text according to your hints.

File: site/metrics.md [hunk 4]
```diff
@@ -20,6 +20,19 @@ exposed:
...
+By default, when started via `weave launch`, weave listens on its local 
+interface to serve metrics. To publish your metrics throughout your cluster, 
+e.g. if your prometheus server is installed on a different host machine,
+you need to set `WEAVE_STATUS_ADDR` to your corresponding IP and port.
+Default port is 6782.
+
+`WEAVE_STATUS_ADDR=X.X.X.X:<IP>`
+
+You can set `WEAVE_STATUS_ADDR=0.0.0.0:6782` to listen on all interfaces, 
+but be aware, this may exposes your metrics to the public internet.
```
Comment (bboreham) [inline site/metrics.md]: s/exposes/expose/

File: site/metrics.md [hunk 5]
```diff
@@ -20,6 +20,19 @@ exposed:
...
 
+#### Publish Router Metrics Endpoint
+
+By default, when started via `weave launch`, weave listens on its local 
+interface to serve metrics. To publish your metrics throughout your cluster, 
+e.g. if your prometheus server is installed on a different host machine,
+you need to set `WEAVE_STATUS_ADDR` to your corresponding IP and port.
+Default port is 6782.
+
+`WEAVE_STATUS_ADDR=X.X.X.X:<IP>`
```
Comment (bboreham) [inline site/metrics.md]: where you have 'IP' I think we'd call it 'port'; also we use angle brackets to denote something a user has to add within a bigger command so just leave those out.

So: 

    ```
    WEAVE_STATUS_ADDR=X.X.X.X:PORT
    ```

**Comment by mmethner** [comment]
Im glad you took the time to point me to the right directions. 
Im aware, that you could easily have changed the docu by yourself in two minutes. Thank you.

I even rebase squashed the four commits into one. Never did this before, but it looks like it worked.

**Comment by bboreham** [comment]
Merged via 7ffd96f4254cdabacea656aefb01f20755ae8b09 - it needed to go on the 1.8 branch to update the docs website

**Comment by bboreham** [comment]
Thanks!

**Pull request closed without merging**